homeassistant/components/prosegur/camera.py

```diff
@@ -0,0 +1,97 @@
+"""Support for Prosegur cameras."""
+from __future__ import annotations
+
+import logging
+
+from pyprosegur.auth import Auth
+from pyprosegur.exceptions import ProsegurException
+from pyprosegur.installation import Camera as InstallationCamera, Installation
+
+from homeassistant.components.camera import Camera
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import (
+    AddEntitiesCallback,
+    async_get_current_platform,
+)
+
+from . import DOMAIN
+from .const import SERVICE_REQUEST_IMAGE
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Prosegur camera platform."""
+
+    platform = async_get_current_platform()
+    platform.async_register_entity_service(
+        SERVICE_REQUEST_IMAGE,
+        {},
+        "async_request_image",
```

Comment:
The camera sends a still picture through an MMS service, so it's not advisable to update periodically and only use it seldomly. 

Prosegur Mobile App also requires this explicit request.

Reply:
Some more explanation:
- The alarm camera will trigger on it's self based on motion sensing and send image to Prosegur server
- User might request an out of order camera capture (that's the service I'm proposing)

This service is therefore independent from update entity service which will periodically get the last camera capture available on Prosegur server.

Reply:
> This service is therefore independent from update entity service which will periodically get the last camera capture available on Prosegur server.

The update entity service allows to request a single update (yes, it could be used periodically in an automation, but by itself it isn't periodically).

Reply:
update_service -> request image from Prosegur server
request_image -> request Prosegur to refresh it's server stored image